src/eval.rs

```diff
@@ -1571,14 +1571,11 @@ fn apply_continuation<F: LurkField>(
                             let scalar_ptr = store
                                 .get_expr_hash(result)
                                 .ok_or_else(|| LurkError::Store("expr hash missing".into()))?;
-                            store.get_char(
-                                char::from_u32(
-                                    scalar_ptr.value().to_u32().ok_or_else(|| {
-                                        LurkError::Eval("Ptr is invalid u32".into())
-                                    })?,
-                                )
-                                .ok_or_else(|| LurkError::Eval("u32 is invalid char".into()))?,
-                            )
+
+                            let res = scalar_ptr.value().to_u32_unchecked();
+                            dbg!(res);
+                            dbg!(char::from_u32(res));
+                            store.get_char(char::from_u32(res).unwrap())
```

Comment:
[nit][id1]: perhaps `.expect("LurkField::to_u32_unchecked invariant violated")` ?

[id1]: ## "personal preference, you can safely ignore"